EventFilter/Phase2PixelRawToDigi/plugins/Phase2ITQCoreProducer.cc

```diff
@@ -0,0 +1,226 @@
+  }  // This will be ignored for 2 chips module
+  else if (row < (kQCoresInChipRow + 2 * kQCoresInChipRowGap)) {
+    updated_row = kQCoresInChipRow;
+  } else {
+    updated_row = (hit.row() - 2 * kQCoresInChipRowGap);
+  }
+
+  // Remapping of the column coordinate
+  if (col < kQCoresInChipColumn) {
+    updated_col = col;
+  } else if (col < kQCoresInChipColumn + kQCoresInChipColumnGap) {
+    updated_col = kQCoresInChipColumn - kQCoresInChipColumnGap;
+  } else if (col < (kQCoresInChipColumn + 2 * kQCoresInChipColumn)) {
+    updated_col = kQCoresInChipColumn;
+  } else {
+    updated_col = (hit.col() - 2 * kQCoresInChipColumnGap);
+  }
+
+  hit.set_row(updated_row);
+  hit.set_col(updated_col);
+
+  return hit;
+}
+
+std::vector<Phase2ITDigiHit> adjustEdges(std::vector<Phase2ITDigiHit> hitList) {
+  /*
+        In-place modification of Hit coordinates to take into account large pixels
+    */
+  std::for_each(hitList.begin(), hitList.end(), &updateHitCoordinatesForLargePixels);
+  return hitList;
+}
+
+std::vector<Phase2ITChip> splitByChip(std::vector<Phase2ITDigiHit> hitList) {
+  // Split the hit list by read out chip
+  std::vector<std::vector<Phase2ITDigiHit>> hits_per_chip(4);
+  for (auto hit : hitList) {
+    int chip_index = (hit.col() < kQCoresInChipColumn) ? 0 : 1;
+    if (hit.row() >= kQCoresInChipRow) {
+      chip_index += 2;
+    }
+    hits_per_chip[chip_index].push_back(hit);
+  }
+
+  // Generate Phase2ITChip objects from the hit lists
+  std::vector<Phase2ITChip> chips;
+  for (int chip_index = 0; chip_index < 4; chip_index++) {
+    chips.push_back(Phase2ITChip(chip_index, hits_per_chip[chip_index]));
+  }
+
+  return chips;
+}
+
+std::vector<Phase2ITChip> processHits(std::vector<Phase2ITDigiHit> hitList) {
+  std::vector<Phase2ITDigiHit> newHitList;
+  newHitList = adjustEdges(hitList);
+
+  std::vector<Phase2ITChip> chips = splitByChip(newHitList);
+  std::vector<bool> code;
+
+  for (size_t i = 0; i < chips.size(); i++) {
+    Phase2ITChip chip = chips[i];
+    code = chip.get_chip_code();
+  }
+
+  return chips;
+}
+
+// ------------ method called to produce the data  ------------
+void Phase2ITQCoreProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  using namespace std;
+
+  unique_ptr<edm::DetSetVector<Phase2ITQCore>> aQCoreVector = make_unique<edm::DetSetVector<Phase2ITQCore>>();
+  unique_ptr<edm::DetSetVector<Phase2ITChipBitStream>> aBitStreamVector =
+      make_unique<edm::DetSetVector<Phase2ITChipBitStream>>();
+
+  auto const& tTopo = iSetup.getData(tTopoToken_);
+
+  edm::Handle<edm::DetSetVector<PixelDigi>> pixelDigiHandle;
+  iEvent.getByToken(pixelDigi_token_, pixelDigiHandle);
+
+  edm::DetSetVector<PixelDigi>::const_iterator iterDet;
+  for (iterDet = pixelDigiHandle->begin(); iterDet != pixelDigiHandle->end(); iterDet++) {
+    DetId tkId = iterDet->id;
+    edm::DetSet<PixelDigi> theDigis = (*pixelDigiHandle)[tkId];
```

Comment:
Range-for would be simpler. Also the `theDigis` can be obtained directly from `iterDet` instead of searching, and it doesn't need to be copied
```suggestion
  for (const auto& theDigis : pixelDigi) {
    DetId tkId = theDigis.id;
```